core/domain/user_jobs_one_off_test.py

```diff
@@ -222,6 +223,84 @@ def test_multiple_users_case(self):
+    """Tests for the one-off username length distribution job."""
+    USER_A_EMAIL = 'a@example.com'
+    USER_A_USERNAME = 'a'
+    USER_A_BIO = 'I am less than 500'
+    USER_B_EMAIL = 'b@example.com'
+    USER_B_USERNAME = 'b'
+    USER_B_BIO = 'Long Bio' *100
+    USER_C_EMAIL = 'c@example.com'
+    USER_C_USERNAME = 'cnone'
+    USER_C_BIO = 'Same Bio' *100
+
+    def _run_one_off_job(self):
+        """Runs the one-off MapReduce job."""
+        job_id = (
+            user_jobs_one_off.UserBioLengthOneOffJob.create_new())
+        user_jobs_one_off.UserBioLengthOneOffJob.enqueue(job_id)
+        self.assertEqual(
+            self.count_jobs_in_taskqueue(
+                queue_name=taskqueue_services.QUEUE_NAME_DEFAULT),
+            1)
+        self.process_and_flush_pending_tasks()
+
+        stringified_output = (
+            user_jobs_one_off.UserBioLengthOneOffJob.get_output(
+                job_id))
+        output = {}
+        #for loop is needed to void `Index out of range` error
+        for stringified_output[0] in stringified_output:
+            output = literal_eval(stringified_output[0])
+        return output
+
+    def test_null_user_bio(self):
+        """Tests the case when userbio is None."""
+        self.signup(self.USER_C_EMAIL, self.USER_C_USERNAME)
+        result = self._run_one_off_job()
+        self.assertEqual(result, {})
+
+    def test_inbound_limit(self):
+        """Tests the case where the userbio is less than 500 characters."""
+        self.signup(self.USER_A_EMAIL, self.USER_A_USERNAME)
+        user_id_a = self.get_user_id_from_email(self.USER_A_EMAIL)
+        user_services.update_user_bio(user_id_a, self.USER_A_BIO)
+        result = self._run_one_off_job()
+        self.assertEqual(result, {})
+
+    def test_outbound_limit(self):
+        """Tests the case where the userbio is more than 500 characters."""
+        self.signup(self.USER_B_EMAIL, self.USER_B_USERNAME)
+        user_id_b = self.get_user_id_from_email(self.USER_B_EMAIL)
+        user_services.update_user_bio(user_id_b, self.USER_B_BIO)
+        len_user_bio_b = len(self.USER_B_BIO)
+        result = self._run_one_off_job()
+        self.assertEqual(result[0], str(len_user_bio_b))
+        self.assertEqual(result[1], list(self.USER_B_USERNAME))
+
+    def test_same_userbio_limit(self):
+        """Tests the case where two users have same userbio limit."""
+        self.signup(self.USER_B_EMAIL, self.USER_B_USERNAME)
+        user_id_b = self.get_user_id_from_email(self.USER_B_EMAIL)
+        user_services.update_user_bio(user_id_b, self.USER_B_BIO)
+        len_user_bio_b = len(self.USER_B_BIO)
+
+        self.signup(self.USER_C_EMAIL, self.USER_C_USERNAME)
+        user_id_c = self.get_user_id_from_email(self.USER_C_EMAIL)
+        user_services.update_user_bio(user_id_c, self.USER_C_BIO)
+        len_user_bio_c = len(self.USER_C_BIO)
+
+        result = self._run_one_off_job()
+        if len_user_bio_b == len_user_bio_c:
```

Comment:
Yep, this can be avoided as you said. Was leaning on changing it but thought would consider your opinion first. 
Will do.